components/logins/src/login.rs

```diff
@@ -2,6 +2,135 @@
+//!
+//!                 XXX TODO: can it be blank? does being blank count as being present for the
+//!                 purposes of conflicting with `formSubmitURL`? does that mean it can be used
+//!                 with any realm?
+//!
+//!                 XXX TODO: we intend to store this as punycode. Should we only accept punycode here?
+//!                 XXX TODO: should we always return this as punycode when reading existing records?
+//!
+//!                 XXX TODO: if neither `httpRealm` or `formSubmitURL` are present, what does that
+//!                 mean for how the login can be used? Does this ever happen in practice?
+//!                 Our docs say "exactly one is allowed to be present" but that doesn't rule out
+//!                 having *neither* of them present.
+//!
+//!  -`usernameField`:  The HTML form field into which the `username` field should be filled, if any.
+//!                     If present, the login should only be used in forms with a corresponding field.
+//!                     There are no restrictions on the contents of this string.
+//!
+//!                 XXX TODO: can this be present if `formSubmitURL` is not set?
+//!
+//!  -`passwordField`:  The HTML form field into which the `password` field should be filled, if any.
+//!                     If present, the login should only be used in forms with a corresponding field.
+//!                     There are no restrictions on the contents of this string.
+//!
+//!                 XXX TODO: can this be present if `formSubmitURL` is not set?
+//!
+//!  - `timesUsed`: A lower bound on the number of times this record as been "used", as an integer.
+//!                 This number may not record uses that occurred on other devices, since some legacy
+//!                 sync clients do not record this information. It may be zero for records obtained
+//!                 via sync that have never been used locally.
+//!
+//!                 This field is managed internally by the logins store and any attempt by the
+//!                 application to specify a value will be ignored.
+//!
+//!                 XXX TODO: are these local machine timestamps, and thus possibly wildly inaccurate?
+//!                 XXX TODO: should we talk about merge strategies here?
+//!
+//!  - `timeCreated`: An upper bound on the time of creation of this login, in integer milliseconds from the unix epoch.
+//!                   This is an upper bound because some legacy sync clients do not record this information;
+//!                   in that case newer clients set `timeCreated` when they see the record for the first time.
+//!
+//!                 This field is managed internally by the logins store and any attempt by the
+//!                 application to specify a value will be ignored.
+//!
+//!                 XXX TODO: are these local machine timestamps, and thus possibly wildly inaccurate?
+//!                 XXX TODO: should we talk about merge strategies here?
```

Comment:
Maybe? Ideally we should take the `min` of the non-zero values to merge.

Reply:
What does a zero in this field indicate?

Reply:
Unfortunately GtiHub is dumb and I no longer can tell which property this was commenting on… I believe this was `timeCreated`… There should never be a 0 here (I guess that should be documented).